tmail-backend/jmap/extensions/src/main/java/com/linagora/tmail/james/jmap/StandaloneEventAttendanceRepository.java

```diff
@@ -0,0 +1,129 @@
+            LOGGER.warn("Defaulting to NeedsAction for message {}", messageId);
+            return AttendanceStatus.NeedsAction;
+        });
+    }
+
+    @Override
+    public Publisher<Void> setAttendanceStatus(Username username, MessageId messageId,
+                                               AttendanceStatus attendanceStatus) {
+        MailboxSession systemMailboxSession = sessionProvider.createSystemSession(username);
+
+        return getFlags(messageId, systemMailboxSession)
+            .flatMap(flags -> {
+                EVENT_ATTENDANCE_FLAGS.forEach(flags::remove);
+                String attendanceFlag =
+                    getFlagFromAttendanceStatus(attendanceStatus).orElseGet(() -> {
+                        LOGGER.warn("No flag found for attendance status {}", attendanceStatus);
+                        LOGGER.warn("Falling back to $needs-action");
+                        return "$needs-action";
+                    });
+                flags.add(attendanceFlag);
+                return doSetFlags(messageId, flags, systemMailboxSession);
+            }).then();
+    }
+
+    private Optional<String> getFlagFromAttendanceStatus(AttendanceStatus attendanceStatus) {
+        return switch (attendanceStatus) {
+            case Accepted -> Optional.of("$accepted");
+            case Declined -> Optional.of("$rejected");
+            case Tentative -> Optional.of("$tentativelyaccepted");
+            case NeedsAction -> Optional.of("$needs-action");
+            case Delegated -> Optional.empty();
+        };
+    }
+
+    private Mono<AttendanceStatus> getAttendanceStatusFromFlags(Flags flags) {
+        long eventAttendanceFlagsCount = Arrays.stream(flags.getUserFlags())
+            .filter(EVENT_ATTENDANCE_FLAGS::contains)
+            .count();
+
+        if (eventAttendanceFlagsCount > 1) {
+            LOGGER.warn("A message should not have more than one event attendance flag");
+        }
+
+        if (flags.contains("$accepted")) {
+            return Mono.just(AttendanceStatus.Accepted);
+        } else if (flags.contains("$rejected")) {
+            return Mono.just(AttendanceStatus.Declined);
+        } else if (flags.contains("$tentativelyaccepted")) {
+            return Mono.just(AttendanceStatus.Tentative);
+        } else if (flags.contains("$needs-action")) {
+            return Mono.just(AttendanceStatus.NeedsAction);
+        } else {
+            return Mono.empty();
+        }
+    }
+
+    private Mono<Void> doSetFlags(MessageId messageId, Flags flagsToSet, MailboxSession session) {
+        return Mono.from(messageIdManager.getMessagesReactive(List.of(messageId), FetchGroup.MINIMAL, session))
+            .map(MessageResult::getMailboxId)
+            .flux()
+            .collectList()
+            .flatMap(mailboxIds ->
+                Mono.from(messageIdManager.setFlagsReactive(
+                    flagsToSet,
+                    MessageManager.FlagsUpdateMode.REPLACE,
```

Comment:
That's the point: I do not want get-then-REPLACE

I want 2 calls to set flags: Add one flag and remove two.